martinLi/assignments/treesBFS/lc107/lc107.js

```diff
@@ -1,30 +1,87 @@
 // Problem Statement #
 
 // Given a binary tree, populate an array to represent its level-by-level traversal in reverse order, i.e., the lowest level comes first. You should populate the values of all nodes in each level from left to right in separate sub-arrays.
+/*
+leaf to root.
+Bottom to top.
+left to right.
+3 
+9  20
+  15 7
+
+
+This would result in the return of
+[15 7 9 20 3] =>([15 7]) ([9 20] [3])
 
-class TreeNode {
 
+*/
+class TreeNode {
   constructor(value) {
     this.value = value;
     this.left = null;
     this.right = null;
   }
-};
+}
+
+const traverse = function (root) {
+  //passing in entire Binary Tree!
+
+  let result = []; // array of arrays
+
+  if (root === null) {
+    //if null then return empty []
+    return result;
+  }
+
+  let queue = [root]; // holds the root val
+  //in queue, what
+  while (queue.length !== 0) {
+    //keep track number of nodes within a queue
+    let node_counter = queue.length;
+    let row_result = [];
+
+    while (node_counter > 0) {
+      //builds up row_result list
+      let current_node = queue.shift();
+      if (current_node.left !== null) {
+        //if it has a left child...
+        queue.push(current_node.left);
+      }
+      if (current_node.right !== null) {
+        //if it has a right child...
+        queue.push(current_node.right);
+      }
+      row_result.push(current_node.value); //we just want to push the node's value
+      node_counter--; //decrement row counter
+    }
+    result.unshift(row_result);
+  }
```

Comment:
Opinionated topic, but, many will say all these single-line comments inside of  code base are better grouped in one section above the code